libclc/utils/libclc-remangler/CMakeLists.txt

```diff
@@ -25,6 +25,24 @@ clang_target_link_libraries(libclc-remangler
   LLVMOption
   )
 
+if(CMAKE_CROSSCOMPILING)
+  if(LLVM_NATIVE_TOOL_DIR AND NOT LIBCLC_REMANGLER_PATH)
+    if(EXISTS "${LLVM_NATIVE_TOOL_DIR}/libclc-remangler${LLVM_HOST_EXECUTABLE_SUFFIX}")
+      set(LLVM_CONFIG_PATH "${LLVM_NATIVE_TOOL_DIR}/libclc-remangler${LLVM_HOST_EXECUTABLE_SUFFIX}")
+    endif()
+  endif()
+
+  if(NOT LIBCLC_REMANGLER_PATH)
+    build_native_tool(libclc-remangler LIBCLC_REMANGLER_PATH)
+    set(LIBCLC_REMANGLER_PATH "${LIBCLC_REMANGLER_PATH}" CACHE STRING "")
+
+    add_custom_target(NativeLibCLCRemangler DEPENDS ${LIBCLC_REMANGLER_PATH})
+    add_dependencies(libclc-remangler NativeLibCLCRemangler)
+  endif()
+else()
+  set(LIBCLC_REMANGLER_PATH "${CMAKE_CURRENT_BINARY_DIR}/bin/libclc-remangler" CACHE STRING "")
+endif()
+
 # If we've not already imported a libclc-remangler tool from an external build,
 # set it up now.
 if(NOT TARGET libclc::libclc-remangler)
```

Comment:
This comment makes me realise this PR breaks the ability to import `libclc-remangler` from another build, using `LIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR`. We should keep that.

Reply:
Yeah, this is actually a bit confusing to me. I found PR #13743 that introduced this, but as far as I can tell, this breaks the `LIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR` functionality since the target is never imported through anything like `find_package`.

I'm realizing now that this might be because I was testing everything with `CMAKE_CROSSCOMPILING` enabled which may have messed with the target import. 

Regardless, the `libclc::libclc-remangler` pattern seems to be unique among the rest of LLVM where other tools used during the build that are also built as a part of LLVM use things like `compile_native_tool` or `setup_host_tool` + `LLVM_NATIVE_TOOL_DIR` which is assigned automatically if needed. 

My latest commit uses `libclc-remangler_target` which is populated by `setup_host_tool`. But I think `LIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR` might still be broken.

I see why `LIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR` is used since it can be built as a standalone. I'm wondering if in the case that it has not been set, then it should be set to `LLVM_NATIVE_TOOL_DIR` if it has been set. Thoughts?